src/main/java/org/joshsim/engine/entity/AttributeManaging.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Interface that defines an entity that can manage attributes, including event handling callbacks.
+ *
+ * @license BSD-3-Clause
+ */
+
+package org.joshsim.engine.entity;
+
+import java.util.HashMap;
+import java.util.Optional;
+import org.joshsim.engine.value.EngineValue;
+
+/**
+ * Interface that defines an entity behavior that can manage attributes, including event handling
+ * callbacks.
+ */
+public interface AttributeManaging {
+
+  /**
+   * Get the name of this type of entity.
+   *
+   * @returns unique name of this entity type.
+   */
+  String getName();
+
+  /**
+   * Get event handlers for all attributes and events.
+   *
+   * @returns Hashmap of (state x attribute x event) to EventHandlerGroups.
+   */
+  HashMap<EventKey, EventHandlerGroup> getEventHandlers();
```

Comment:
(Optional) I like EventKey! I wonder about exposing this object (single parameter to `getEventHandler`?)